benchmarks/notesize.py

```diff
@@ -0,0 +1,111 @@
+"""
+How big is a note? Do a bunch of allocations and plot memory usage.
+"""
+# fmt: off
+# flake8: noqa E129,E303
+
+__author__ = "Roger B. Dannenberg"
+
+import gc
+import sys
+import tracemalloc
+
+import matplotlib.pyplot as plt
+import psutil
+
+# from memory_profiler import memory_usage
+from amads.core import Note
+
+"""
+How big is a note?
+"""
+import sys
+
+
+def get_total_size(obj, seen=None):
+    """Recursively find the total memory size of an object, including its attributes."""
```

Comment:
I'm sure people have created specialised packages for quantifying object sizes, isn't it going to be errorprone to write your own? e.g. https://pypi.org/project/objsize/